src/frameworks/sequelize/templates/model/modelClass.ts

```diff
@@ -113,12 +113,14 @@ const classFieldType = (
   const fieldType = creationOptional
     ? `CreationOptional<${tsType}>`
     : required
-    ? tsType
-    : `${tsType} | null`
+      ? tsType
+      : `${tsType} | null`
+
+  const isOptional = required ? '' : '?';
 
   return [
     noSupportedDetails(type, dbOptions.sqlDialect),
-    `${comment}declare ${camelCase(name)}: ${fieldType}`,
+    `${comment}declare ${camelCase(name)}${isOptional}: ${fieldType}`,
```

Comment:
The TS code generated by Sequelize UI follow the newer [Sequelize TypeScript conventions](https://sequelize.org/docs/v6/other-topics/typescript/) which make `?:` properties unnecessary. `InferCreationAttributes` will make automatically make any nullable or `CreationOptional` fields `?:` for the `build` and `create` arguments (via the `CreationAttributes` utility type). For example, let's say we have a model `Post` where the fields `title`, `slug` and `published` are required, and all fields are optional. We'd generate the following model:

```ts
import {
  CreationOptional,
  DataTypes,
  InferAttributes,
  InferCreationAttributes,
  Model,
  Sequelize,
} from "sequelize/types";

export class Post extends Model<
  InferAttributes<Post>,
  InferCreationAttributes<Post>
> {
  declare id: CreationOptional<number>;
  declare title: string;
  declare metaTitle: string | null;
  declare slug: string;
  declare summary: string | null;
  declare published: boolean;
  declare publishedAt: Date | null;
  declare content: string | null;
  declare createdAt: CreationOptional<Date>;
  declare updatedAt: CreationOptional<Date>;

  static initModel(sequelize: Sequelize): typeof Post {
    Post.init(
      {
        id: {
          type: DataTypes.INTEGER.UNSIGNED,
          primaryKey: true,
          autoIncrement: true,
          allowNull: false,
        },
        title: {
          type: DataTypes.STRING(75),
          allowNull: false,
        },
        metaTitle: {
          type: DataTypes.STRING(100),
        },
        slug: {
          type: DataTypes.STRING(100),
          allowNull: false,
          unique: true,
        },
        summary: {
          type: DataTypes.TEXT,
        },
        published: {
          type: DataTypes.BOOLEAN,
          allowNull: false,
        },
        publishedAt: {
          type: DataTypes.DATE,
          allowNull: false
        },
        content: {
          type: DataTypes.TEXT,
        },
        createdAt: {
          type: DataTypes.DATE,
        },
        updatedAt: {
          type: DataTypes.DATE,
        },
      },
      {
        sequelize,
      }
    );

    return Post;
  }
}
```

The first argument to `Post.create` or `Post.build` is `CreationAttributes<Post>`, which is more or less equivalent to:
```ts
{
    title: string;
    metaTitle?: string | null;
    slug: string;
    summary?: string | null;
    published: boolean;
    publishedAt?: Date | null;
    content?: string | null;
}
```

so all of the following examples would compile properly:

```ts
Post.build({ title: 'foo', slug: 'foo', published: false })
Post.create({ title: 'foo', slug: 'foo', published: false })
Post.build({ title: 'foo', slug: 'foo', published: false, metaTitle: null })
Post.create({ title: 'foo', slug: 'foo', published: false, metaTitle: undefined })
```

It doesn't make sense to have the class properties types allow `undefined` because the return type of `create`, `build`, `update`, or any of the query functions will never have these fields as `undefined`.

One enhancement here that I think _does_ make sense would be to add `CreationOptional` for any fields with a default value.